src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -151,57 +151,68 @@ MatrixXd Segment::Solution::getDynamicsContribution(
         );
-
-        dynamicsContributionMatrix.row(stateIndex) = dynamicsContributionAtState;
+        dynamicsContributions.add(dynamicsContributionAtState);
     }
 
-    return dynamicsContributionMatrix;
+    return dynamicsContributions;
 }
 
-MatrixXd Segment::Solution::getDynamicsAccelerationContribution(
+Array<Vector3d> Segment::Solution::getDynamicsAccelerationContributions(
     const Shared<Dynamics>& aDynamicsSPtr, const Shared<const Frame>& aFrameSPtr
 ) const
 {
-    return this->getDynamicsContribution(aDynamicsSPtr, aFrameSPtr, {CartesianVelocity::Default()});
+    const Array<VectorXd> dynamicsAccelerationContributions =
+        this->getDynamicsContributions(aDynamicsSPtr, aFrameSPtr, {CartesianVelocity::Default()});
+
+    Array<Vector3d> accelerationContributions = Array<Vector3d>::Empty();
+    for (const VectorXd& dynamicsAccelerationContribution : dynamicsAccelerationContributions)
+    {
+        accelerationContributions.add(dynamicsAccelerationContribution.head<3>());
+    }
+
+    return accelerationContributions;
 }
 
-Map<Shared<Dynamics>, MatrixXd> Segment::Solution::getAllDynamicsContributions(const Shared<const Frame>& aFrameSPtr
+Array<Map<Shared<Dynamics>, VectorXd>> Segment::Solution::getAllDynamicsContributions(
+    const Shared<const Frame>& aFrameSPtr
 ) const
 {
     // TBI: Use smart caching for multiple calls in the future
 
-    // Each MatrixXd contains the contribution of a single dynamics for all the segment states
-    Map<Shared<Dynamics>, MatrixXd> dynamicsContributionsMap = Map<Shared<Dynamics>, MatrixXd>();
+    Map<Shared<Dynamics>, Array<VectorXd>> dynamicsContributionsMap = Map<Shared<Dynamics>, Array<VectorXd>>();
 
+    // Create map for each dynamics
     for (const Shared<Dynamics>& aDynamicsSPtr : this->dynamics)
     {
-        MatrixXd dynamicsContribution = this->getDynamicsContribution(aDynamicsSPtr, aFrameSPtr);
-        dynamicsContributionsMap.emplace(aDynamicsSPtr, dynamicsContribution);
+        Array<VectorXd> dynamicsContributions = this->getDynamicsContributions(aDynamicsSPtr, aFrameSPtr);
+        dynamicsContributionsMap.emplace(aDynamicsSPtr, dynamicsContributions);
+    }
+
+    Array<Map<Shared<Dynamics>, VectorXd>> dynamicsContributionsMaps = Array<Map<Shared<Dynamics>, VectorXd>>::Empty();
+
+    const Size numberOfstates = this->states.getSize();
+    for (Index stateIndex = 0; stateIndex < numberOfstates; ++stateIndex)
+    {
+        // Create map for each state and fill it with the dynamics contributions
+        Map<Shared<Dynamics>, VectorXd> dynamicsContributionMapAtState = Map<Shared<Dynamics>, VectorXd>();
+        for (const Shared<Dynamics>& aDynamicsSPtr : this->dynamics)
+        {
+            dynamicsContributionMapAtState.emplace(aDynamicsSPtr, dynamicsContributionsMap[aDynamicsSPtr][stateIndex]);
+        }
+        dynamicsContributionsMaps.add(dynamicsContributionMapAtState);
```

Comment:
I feel like this will duplicate so much data in terms of repeated headers.
Can you provide an example that this data format is cleaner?
Currently:
```
{
    CartesianVelocity : [[0,1,2], [2,3,4], [4,5,6]...]
    Acceleration: [[5,6,7], [6,7,8], [7,8,9]]
}
```
vs.
- {CartesianVelocity: [0,1,2], Acceleration: [4,5,6]}
- {CartesianVelocity: [0,1,2], Acceleration: [4,5,6]}
- {CartesianVelocity: [0,1,2], Acceleration: [4,5,6]}
...